_posts/2024-12-19-cosmicqc.md

```diff
@@ -0,0 +1,285 @@
+Single-cell quality control for morphology profiles is not has well developed in the image-based profiling realm.
+
+I will preface this by saying that you are currently able to filter objects in CellProfiler using morphology measurements, but the module isn't very intuitive.
+You can set limits for what measurement is considered good or bad, but how would you know what the limits of a feature is without exporting the data and performing analyses to get a sense of the range of values for your dataset.
+In my opinion, this module is good to have, but it isn't a solution to single-cell quality control.
+
+I did conduct some research into seeing if there were already methods out there to perform this task.
+I was only able to identify one paper that makes an attempt at this process.
+The publication in BMC Bioinformatics is called "[A cell-level quality control workflow for high-throughput image analysis](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03603-5)", and it was published back in 2020.
+In the background section, they state:
+
+> *"To our knowledge, this represents the first QC method and scoring metric that operate at the cell level."*
+
+Since I couldn't find any other paper that focuses on quality control at the single-cell level, I completely agree with their statement. 
+To summarize the paper, they present a cell-level quality control workflow which uses machine learning to detect artifacts within images.
+They utilized CellProfiler to segment single cells and extract features from both the cells and whole images.
+They wanted to group images based on their phenotype, so they utilized CellProfiler features (like `PowerLogLogSlope`, `FocusScore`, `MeanIntensity`, `Cell Count`, etc.) to detect images that contains artifacts versus no artifacts.
+Once images were grouped together, they performed a method using KDE to sample the images into their respective phenotypes.
+They then trained an SVM (Support Vector Machine) per phenotype group, to classify where there are artifacts and where there are cells.
+I believe I barely scratch the surface of the complexities behind this method, so feel free to read the paper if you want a more in-depth look.
+
+What I really liked about this paper is that it is attempting to preserve images that include artifact, as these images still contain other cells.
+It is different than the current process where whole images are removed that fail QC metrics like blur and over-saturation (traditional done in CellProfiler).
+But, I also have my suspicions about using cells that come from an image with corruption, such as an artifact.
+I wonder how much things like an artifact or smudge could be directly impacting the biology the cells.
+In my opinion, I would rather default to losing those cells in an image than risk any issues in downstream interpretation of the results do to any impacts the artifacts in an image had.
```

Comment:
```suggestion
This would be a cool research project, but, in my opinion, I would rather lose cells in an image with an artifact than risk any issues in downstream interpretation of the results that might be due to any impacts the artifacts in the image had.
```

Reply:
updated!